### PR TITLE
[QC PART 4] Add notebook exploring lowly expressed genes

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
_I'm breaking up the files from [`jaclyn-taroni/qc_viz`]( https://github.com/AlexsLemonade/compendium-processing/tree/jaclyn-taroni/qc_viz) into smaller PRs. (That branch will be retained for posterity.)_

We observed some differences between platforms (#17). It could be the case that genes that are often zero in RNA-seq data tend to be lowly expressed (and therefore will have low average expression values on microarrays) and this may contribute to the difference. This notebook explores this notion of lowly expressed genes. We observed a pattern that is consistent with what we stated above.

### CODE REVIEW

File: quality_check/06-lowly_expressed_genes.Rmd [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+```{r}
+seq_col_modes <- apply(seq_mat, 2, calculate_mode)
+```
+
+Find the indices of each of the "used-to-be-zero" values
+
+```{r}
+zero_gene_list <- 
+  lapply(1:ncol(seq_mat),
```
[the remark this answers is not in the document]
Comment (cansav09) [reply quality_check/06-lowly_expressed_genes.Rmd]: May also be able to combine this into the apply you are doing above. 
I think there's a more easy to follow/streamlined way to do this step. IDK. Hard to say without tme playing around with it more.

File: quality_check/06-lowly_expressed_genes.Rmd [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+`%>%` <- dplyr::`%>%`
+```
+
+Function for calculating the mode from 
+https://stackoverflow.com/questions/2547402/is-there-a-built-in-function-for-finding-the-mode
+
+```{r}
+calculate_mode <- function(x) {
+  ux <- unique(x)
+  ux[which.max(tabulate(match(x, ux)))]
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply quality_check/06-lowly_expressed_genes.Rmd]: We want this to return the value of the mode, which is why we need `which.max`. There may be another what to do it but returning `max(tabulate(match(x, ux)))` does not give me the desired behavior.

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply quality_check/06-lowly_expressed_genes.Rmd]: I've added a comment about the usage.

File: quality_check/06-lowly_expressed_genes.Rmd [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+## Explore distributions
+
+This test compendium is quantile normalized (QN'd). 
+RNA-seq data, even after all the other processing steps, likely has many 
+zeroes and we can spot that in the data post-QN.
+Let's look at the distribution of a single RNA-seq sample for illustrative
+purposes.
+
+```{r}
+example_ecdf_plot <- data.frame(x = seq_mat[, 34]) %>%
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply quality_check/06-lowly_expressed_genes.Rmd]: No particular reason, I'll do what you suggested below.

File: quality_check/06-lowly_expressed_genes.Rmd [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+calculate_mode <- function(x) {
+  ux <- unique(x)
+  ux[which.max(tabulate(match(x, ux)))]
+}
+```
+
+## Read in data
+
+```{r}
+test_compendium_dir <- file.path("refinebio-data", "larger-compendium",
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply quality_check/06-lowly_expressed_genes.Rmd]: I have an RDS version of this from notebook `05` actually. That's way faster to use!

File: quality_check/06-lowly_expressed_genes.Rmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply quality_check/06-lowly_expressed_genes.Rmd]: I need information about the `rownames` -- how would I do this with `apply`? 
For example:
```R
> apply(seq_mat[, 1:6], 2, function(x) message(rownames(x)))






NULL

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply quality_check/06-lowly_expressed_genes.Rmd]: Can do:
```R
zero_gene_list <- apply(seq_mat, 2, 
                        function(x) { 
                          mode_value <- calculate_mode(x)
                          mode_index <- which(x == mode_value)
                          rownames(seq_mat)[mode_index]
                        })
```
But still returns a list

File: quality_check/06-lowly_expressed_genes.Rmd [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+zero_gene_list <- 
+  lapply(1:ncol(seq_mat),
+         function(x) rownames(seq_mat)[which(seq_mat[, x] == seq_col_modes[x])])
+```
+
+Is it the same genes over and over again? 
+Let's count.
+
+```{r}
+zero_counts_per_gene <- data.frame(table(unlist(zero_gene_list)))
```
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply quality_check/06-lowly_expressed_genes.Rmd]: Even with the change to `apply`, this will be a list.

**Pull request merged**